### PR TITLE
Improve trace tags in ingress and proxy spans

### PR DESCRIPTION
**Pull request by slok**
Following the [semantics of Opentracing](https://github.com/opentracing/specification/blob/master/semantic_conventions.md) and trying to add more information to traces I have add and modified some tags and logs on proxy and ingress spans.

* Add http host (I think that this tag could be very interesting when querying traces in some cases).
* Add http path tag.
* Add remote address tag.
* Modify http url tag to set the full url (when available).
* Set server span kind on ingress span (important to know if the http data is from an ingress request or a egress one).
* Set component to `skipper`.
* Set error log on proxy span to follow the semantic convention of errors.

Example:

example trace

### CODE REVIEW

File: proxy/proxy.go [hunk 1]
```diff
@@ -649,13 +649,12 @@ func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, *proxyError) {
 	} else {
 		proxySpan = p.openTracer.StartSpan("proxy", ot.ChildOf(ingress.Context()))
 	}
-	ext.SpanKind.Set(proxySpan, ext.SpanKindRPCClientEnum)
+	defer proxySpan.Finish()
+	ext.SpanKindRPCClient.Set(proxySpan)
```
[the remark this answers is not in the document]
Comment (slok) [reply proxy/proxy.go]: I don't know if I understood correctly the question but I will answer what I understood.

Yes, it should, from the code, a new span is created from the parent one (`ingress` span). I have check how is shown:

**Comment by szuecs** [comment]
:+1:

File: proxy/proxy.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (szuecs) [reply proxy/proxy.go]: Thanks, exactly what I wanted to see. 
Super happy to see this improvement.

**Comment by lmineiro** [comment]
:+1:

File: proxy/proxy.go [hunk 2]
```diff
@@ -1009,3 +1011,12 @@ func (p *Proxy) Close() error {
 	close(p.quit)
 	return nil
 }
+
+func (p *Proxy) setCommonSpanInfo(u *url.URL, r *http.Request, s ot.Span) {
+	ext.Component.Set(s, "skipper")
```
Comment (lmineiro) [inline proxy/proxy.go]: This seems unnecessary/redundant with the service name or component name provided initially to the Tracers. I'd use it if we intend to have more fine-grained components like `skipper:proxy`, `skipper:dataclient` or similar.

Comment (slok) [reply proxy/proxy.go]: I think that `component` tag doesn't specify what is the nature of the application but the technology itself, your tracer could be named `skipper`, `skipper-proxy`, `skipper-myapp`... but the app is a `skipper`.

From the [conventions](https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table)
> component (string): The software package, framework, library, or module that generated the associated Span. E.g., "grpc", "django", "JDBI".

**Comment by vetinari** [comment]
:+1:

**Comment by slok** [comment]
Thank you! :)

**Comment by szuecs** [comment]
@slok we have to thank you!

**Pull request merged**